content/articles/introduction-to-hibernate/index.md

```diff
@@ -0,0 +1,280 @@
+
+#### Hibernate Example
+Here's an example of  implementation of Hibernate.It is for one-to-one biderectional mapping.
+
+```java
+ //This is customer.java class
+import java.io.Serializable;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+
+@Entity
+public class customer implements Serializable {
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+
+	@Column(name = "customer_name")
+	private String customer_name;
+
+	@Column(name = "custmer_id")
+	private String custmer_id;
+
+	@Column(name = "customer_email")
+	private customer_email;
+
+	@OneToOne(cascade = CascadeType.ALL)
+	@JoinColumn(name = "location_id", referencedColumnName = "id")
+	private Location location;
+	
+}
+//This is location.java class
+//import  the same packages  as the one imported in customer.java class
+@Entity
+public class Location {
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+
+	@Column(name = "state")
+	private String state;
+
+	@Column(name = "state_number")
+	private String state_number;
+
+	@OneToOne(cascade = CascadeType.ALL, mappedBy = "location")
+	private Customer customer;
+
+}
+//main class
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.hibernate.SessionFactory;
+import java.util.List;
+import org.hibernate.cfg.Configuration;
+import com.netsurfingzone.entity.Location;
+import com.netsurfingzone.entity.Customer;
+
+public class Main {
+	public static void main(String[] args) {
+		SessionFactory factory = null;
+		Session session = null;
+		Configuration configuration = new Configuration().configure();
+		try {
+
+			factory = configuration.buildSessionFactory();
+			session = factory.openSession();
+			Transaction k = session.beginTransaction();
+			
+			
+			Customer customer1 = new Customer();
+			customer1.setCustomerName("Kelvin");
+			customer1.setCustomerId("56");
+			customer1.setCustomerEmail("kelvin@gmail.com");
+			
+			Customer customer2 = new Customer();
+			customer2.setCustomerName("Joy");
+			customer2.setCustomerId("26");
+			customer2.setCustomerEmail("joy@gmail.com");
+			
+			Location location1 = new Location();
+			location1.setState("Turs");
+			location1.setStateNumber(51");
+			Location location2 = new Location();
+			location2.setState("Rouds");
+			location2.setStateNumber(45");
+			
+			
+			customer1.setLocation(location1);
+			customer2.setLocation(location2);
+			session.save(customer1);
+			session.save(customer2);
+			
+			
+			k.commit();
+		} catch (Exception e) {
+			e.printStackTrace();
+
+		} finally {
+            factory.close();
+			session.close();
+			
+		}
+	}
+
+}

```

Comment:
Please explain the above code

Reply:
@WanjaMIKE,I am done with explaining the code.

Reply:
@WanjaMIKE ,I am done with explaining the code